### ISSUE TITLE
kwargs inside the decorator are wrong

### ISSUE THREAD

**Issue by mxposed** [micheles/decorator#103]
Hello,

I ran into a issue when I try to assess certain kwargs inside the decorator. In my situation I had one kwarg passed to the function, but it was inside `args` inside the decorator. Am I doing something wrong or is this a bug or is this a feature?

The test below fails for me and prints `args (True,) kwargs {}`

```python
def test_decorator_kwargs():
    @decorator.decorator
    def loader(func, *args, **kwargs):
        print(f"args {args} kwargs {kwargs}")
        return kwargs

    @loader
    def my_loader(test=False):
        return test

    kwargs = my_loader(test=True)
    assert "test" in kwargs
    assert kwargs["test"]

    kwargs = my_loader(test=False)
    assert "test" in kwargs
    assert not kwargs["test"]
```

I'm using decorator 4.4.2 on python 3.7.1

**Comment by bauman** [comment]
hah, I'm having the opposite problem with #104 


I think this behavior is correct of 4.4.2 is correct.  

in your example `func(test=False)` 


`test` is a named positional argument, not a keyword argument.  

I totally understand that those two things _SHOULD_ be the same thing, but python used to (?) treat them differently.

**Comment by bauman** [comment]
nuts, but I think decorator v4 more strictly adhered to python's implementation (for better or worse)

If the decorated function uses a variable as a named positional variable, it isn't included in the kwargs dict.

```python
def test_behavior(*args, **kwargs):
    print(f"args: {args} \n kwargs:{kwargs}")

def test_behavior2(required, optional=None, **kwargs):
     print(f"req: {required} \noptional={optional} \nkwargs:{kwargs}")


test_behavior()
args: () 
 kwargs:{}
test_behavior("hi")
args: ('hi',) 
 kwargs:{}
test_behavior(named="hi")
args: () 
 kwargs:{'named': 'hi'}



test_behavior2("hi")
req: hi 
optional=None 
kwargs:{}
test_behavior2("hi", "world")
req: hi 
optional=world 
kwargs:{}
test_behavior2("hi", optional="world" )
req: hi 
optional=world 
kwargs:{}
test_behavior2("hi", optional="world", something_else="weird")
req: hi 
optional=world 
kwargs:{'something_else': 'weird'}
```

**Comment by mxposed** [comment]
@bauman you may be right in how python names them, but I'm not sure I agree semantically.

Consider this:
```python
def test(x=1, y=2):
  print(f"x={x}, y={y}")
```

While it is true that I cannot do `test(10, x=20)` but I can do `test(y=10, x=20)`.

Anyway, my usecase is this: the decorator needs to check an environment variable and already supplied named arguments, and if either of them has indicated that this is a test, then set the `test` named argument if the underlying function has that named argument. Current behaviour makes it quite hard to check if the supplied named arguments had `test` already, but maybe I just am doing it wrong. I suppose I can look at the inspection of the underlying function, get the position number of the named argument and then ask the `args` for that value

**Comment by bauman** [comment]
I completely agree with you @mxposed 

I don't think that the problem we're talking about is this maintainers problem though.

Python itself invented these three different types of arguments where positional and named positional are loosely used in the same `args` and this other thing that semantically __is__ a named argument is something different in a keyword arg to python. 

perhaps we in the python community have been abusing this and should have chosen between positional arguments and keyword arguments and left the named positional defaultable arguments alone.

**Comment by micheles** [comment]
I knew this was going to bite me. See my last comment in https://github.com/micheles/decorator/issues/39.
I don't think this is fixable, unless I go back to using "exec", which I do not want to do. Also notice the sentence on the documentation **lately I have come to believe that decorating functions with keyword arguments is not such a good idea, and you may want not to do that**. I need to sleep on this issue to see what can be done. For the moment people with breakage should not upgrade to version 5.

**Comment by micheles** [comment]
@mxposed 
The behavior that you want is now available in decorator 5.0.5:
```python
from decorator import decorator

def loader(func, *args, **kwargs):
    print(f"{args=} {kwargs=}")
    return kwargs

@decorator(loader, kwsyntax=True)
def my_loader(test=False):
    return test

my_loader(test=True)  # args=() kwargs={'test': True}
my_loader(test=False) # args=() kwargs={'test': False}
```
Notice that you can get what you want also with the old decorator module, but you need to use kw-only-arguments, i.e.
write my_loader as follows:
```python
@loader
def my_loader(*, test=False):
    return test
```

**Comment by mxposed** [comment]
Thank you! Happy Easter!
